var/spack/repos/builtin/packages/starpu/package.py

```diff
@@ -0,0 +1,148 @@
+    depends_on("fxt", when='+fxt')
+    depends_on("simgrid", when='+simgrid')
+    depends_on("simgrid+smpi", when='+simgrid+mpi')
+    depends_on("simgrid+mc", when='+simgridmc')
+
+    conflicts('+shared', when='+mpi+simgrid', msg="Simgrid MPI require to disable shared library")
+
+    def autoreconf(self, spec, prefix):
+        if not os.path.isfile("./configure"):
+            autogen = Executable("./autogen.sh")
+            autogen()
+
+    def configure_args(self):
+        spec = self.spec
+
+        config_args = [
+            '--disable-build-doc',
+            '--enable-blas-lib=none',
+            '--disable-mlr',
+        ]
+
+        # add missing lib for simgrid static compilation,
+        # already fixed since StarPU 1.2.1
+        if spec.satisfies('+fxt'):
+            mf = FileFilter('configure')
+            mf.filter('libfxt.a -lrt', 'libfxt.a -lrt -lbfd')
+
+        mpicc = ""
+        if spec.satisfies('+mpi~simgrid'):
+            mpicc = spec['mpi'].mpicc
+        elif spec.satisfies('+mpi+simgrid'):
+            mpicc = "%s/bin/smpicc" % spec['simgrid'].prefix
+
+        config_args.extend([
+            "--%s-shared"         % ('enable' if '+shared'   in spec else 'disable'),
+            "--%s-debug"          % ('enable' if '+debug'    in spec else 'disable'),
+            "--%s-verbose"        % ('enable' if '+verbose'  in spec else 'disable'),
+            "--%s-fast"           % ('enable' if '+fast'     in spec else 'disable'),
+
+            "--%s-build-tests"    % ('enable' if '+examples' in spec else 'disable'),
+            "--%s-build-examples" % ('enable' if '+examples' in spec else 'disable'),
+
+            "--%s-fortran"        % ('enable' if '+fortran'  in spec else 'disable'),
+            "--%s-openmp"         % ('enable' if '+openmp'   in spec else 'disable'),
+
+            "--%s-opencl"         % ('disable' if '~opencl' in spec
+                                     or '+simgrid' in spec else 'enable'),
+            "--%s-cuda"           % ('disable' if '~cuda'   in spec
+                                     or '+simgrid' in spec else 'enable'),
+
+            "--disable-mpi"       if '~mpi'  in spec else "--enable-mpi",
+            "--without-mpicc"     if '~mpi'  in spec else "--with-mpicc=%s" % mpicc,
+
+            "--with-hwloc=%s"     % spec['hwloc'].prefix,
+        ])
+
+        if spec.satisfies('+fxt'):
+            config_args.append("--with-fxt=%s" % spec['fxt'].prefix)
+            if spec.satisfies('@1.2:') or spec.satisfies('@git-1.2'):
+                config_args.append("--enable-paje-codelet-details")
+
+        if spec.satisfies('+simgrid'):
+            config_args.append("--enable-simgrid")
+            config_args.append("--with-simgrid-dir=%s" % spec['simgrid'].prefix)
+            if spec.satisfies('+simgridmc'):
+                config_args.append("--enable-simgrid-mc")
+
+        # On OSX, deactivate glpk
+        if platform.system() == 'Darwin':
```

Comment:
Check the spec instead of asking Python: `self.spec.satisfies('platform=darwin')`. See e.g. `brotli/package.py`.

Reply:
fixed